pypeit/core/extract.py

```diff
@@ -2060,15 +2129,19 @@ def ech_objfind(image, ivar, slitmask, slit_left, slit_righ, order_vec, maskslit
             # ToDO assign this to sobjs_align for use in the extraction
             SNR_arr[iord,iobj] = med_sn
             sobjs_align[indx][0].ech_snr = med_sn
+            # For hand extractions
+            slitfracpos_arr[iord,iobj] = sobjs_align[indx][0].SPAT_FRACPOS
 
     # Purge objects with low SNR that don't show up in enough orders, sort the list of objects with respect to obj_id
     # and orderindx
     keep_obj = np.zeros(nobj,dtype=bool)
     sobjs_trim = specobjs.SpecObjs()
     # objids are 1 based so that we can easily asign the negative to negative objects
     iobj_keep = 1
+    hand_frac = [-1000] if hand_extract_dict is None else [int(np.round(ispat*1000)) for ispat in f_spats]
```

Comment:
These are ugly and dangerous hacks. Touch base with me about how to implement this correctly.

Reply:
TODO ADDED.